### PR TITLE
Update planned enhancements

### PR DESCRIPTION
**Pull request by jktang14**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -459,9 +448,20 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 * **Private contact detail**: A contact detail that is not meant to be shared with others
 * **Versioned AddressBook**: An extension to the AddressBook class that allows undoing/redoing actions.
 
-
 --------------------------------------------------------------------------------------------------------------------
 
+## **Appendix: Planned Enhancements**
+**Note:** Our team size is 5
+
+1. **Make GUI wrap large addresses to the next line:** With the current implementation, if you type in an address which is longer than the width of the screen, it stretches the person card (for all contacts), forcing the user to scroll to the right to see the rest of the address, the project status and deadline. For future implementation, we plan to wrap the address, such that if it is very long, it wraps to the next line.
```
Comment (tanwartushar) [inline docs/DeveloperGuide.md:456]:
```suggestion
1. **Make the GUI wrap large addresses to the next line:** With the current implementation, if you type in an address which is longer than the width of the screen, it stretches the person card (for all contacts), forcing the user to scroll to the right to see the rest of the address, the project status and deadline. In future updates, we plan to enable `textWrap` and set a `maxWidth` for the address label to enhance the UI and improve readability.
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -459,6 +448,19 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+5. **Adjust sorting function to be stable:** With the current implementation of sort, a rather rare bug occurs. When you first sort the list, clients will be sorted correctly based on name/deadline depending on the parameter you chose. Following this, the bug arises when multiple clients have the same name (if you sorted based on name) or same deadline (if you sorted based on deadline). If you change one of the fields of the people with the same name (if you sorted based on name) or same deadline (if you sorted based on deadline), sometimes, the list will reorder among people of the same name/deadline. For future implementations, we plan to modify this such that the list doesn't reorder those with the same name/deadline.
+6. **Make add and edit command accept blacklisted client status:** The current implementation throws an error of invalid command format for the parameter cs/blacklisted. Users can work around this by using the blacklist command implemented. In the future, we plan to remove the restriction on blacklisted client status from both AddCommandParser and EditCommandParser.
```
Comment (Dinoman44) [inline docs/DeveloperGuide.md:461]:
```suggestion
6. **Make `add` and `edit` commands accept blacklisted client status:** The current implementation throws an error of invalid command format for the parameter `cs/blacklisted`. Users can work around this by using the blacklist command implemented. In the future, we plan to remove the restriction on blacklisted client status from both `AddCommandParser` and `EditCommandParser`.
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -459,6 +448,19 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+7. **Make rejection of blacklisted client status for add and edit command more specific:** This is in relation to point 5. The current error message thrown is that of **Invalid command format!** which does not accurately explain to the user that the issue lies with the blacklisted parameter. We plan to make the error message mention this `cs/blacklisted cannot be used for this command` **It is important to note that this enhancement will be deemed redundant once point 5. has been implemented, where cs/blacklisted is indeed accepted.
```
Comment (Dinoman44) [inline docs/DeveloperGuide.md:462]:
```suggestion
7. **Make rejection of blacklisted client status for add and edit command more specific:** This is an alternative to point 6 above. The current error message thrown is that of **"Invalid command format!"** which does not accurately explain to the user that the issue lies with the blacklisted parameter. We plan to make the error message mention that `cs/blacklisted cannot be used for this command`. **It is important to note that this enhancement will be deemed redundant if point 6 is been implemented, where `cs/blacklisted` is indeed accepted.**
```

File: docs/UserGuide.md [hunk 4]
```diff
@@ -206,7 +206,7 @@ Format: `find [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [d/DEADLINE] [t/TA
 * Phone number, email, address, project status, tags, payment status, client status and deadline must match the exact string.
   e.g. `ps/in progress` will not match `ps/in prog`.
 * All fields need to be matched for a person to be found
-  e.g. `find n/Rah py/paid ps/in progress` matches those clients whose name starts with "Rah", payment status is `paid` and project status is `in progress`.
+  e.g. `find n/Rah py/paid ps/in progress` matches those clients whose name segment (first name, middle name, last name etc) starts with "Rah", payment status is `paid` and project status is `in progress`.
```
Comment (Dinoman44) [inline docs/UserGuide.md:209]: Sorry about that, good catch

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -459,6 +448,19 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+4. **Allow users to clear current sorting scheme:** Currently, users cannot 'undo' a sort command. In the future, we plan to add an undo and redo feature, which will complete this enhancement as well. Do refer to [the proposed undo and redo feature](#proposed-undoredo-feature).
+5. **Adjust sorting function to be stable:** With the current implementation of sort, a rather rare bug occurs. When you first sort the list, clients will be sorted correctly based on name/deadline depending on the parameter you chose. Following this, the bug arises when multiple clients have the same name (if you sorted based on name) or same deadline (if you sorted based on deadline). If you change one of the fields of the people with the same name (if you sorted based on name) or same deadline (if you sorted based on deadline), sometimes, the list will reorder among people of the same name/deadline. For future implementations, we plan to modify this such that the list doesn't reorder those with the same name/deadline.
```
Comment (Dinoman44) [inline docs/DeveloperGuide.md:460]: Can also mention that this is an issue with the `ObservableList` interface (or whatever its called, I forget) we are using, and add a hyperlink this post for the issue: [https://bugs.openjdk.org/browse/JDK-8301761](https://bugs.openjdk.org/browse/JDK-8301761)

**Review by Dinoman44 — COMMENTED** [review]
LGTM, just a few edits to make

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -459,6 +448,19 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 * **Versioned AddressBook**: An extension to the AddressBook class that allows undoing/redoing actions.
 
+--------------------------------------------------------------------------------------------------------------------
+
+## **Appendix: Planned Enhancements**
+**Note:** Our team size is 5
+
+1. **Make the GUI wrap large addresses to the next line:** With the current implementation, if you type in an address which is longer than the width of the screen, it stretches the person card (for all contacts), forcing the user to scroll to the right to see the rest of the address, the project status and deadline. In future updates, we plan to enable `textWrap` and set a `maxWidth` for the address label to enhance the UI and improve readability.
+2. **Add support for patronymics, like 's/o' or 'd/o', in the name field:** The current implementation only allows alphanumeric characters, spaces, and dashes, and so users have to work around this restriction by using "s o" or "son of". In the future, we will adjust the regex to something like this `[\\p{Alnum}]+(([ -][\\p{Alnum}]+)*|(([ -](s|d)/o\\s)[\\p{Alnum}]+))\\s*` to allow special strings like "s/o" and "d/o".
```
[the remark this answers is not in the document]
Comment (jktang14) [reply docs/DeveloperGuide.md:457]: LOL

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (jktang14) [reply docs/DeveloperGuide.md:460]: Yep this is under rahul's PR

Comment (rahula1008) [reply docs/DeveloperGuide.md:460]: I think my PR addresses this

**Pull request merged**